README.md

```diff
@@ -1,11 +1,222 @@
+HTTP/1.1 200 OK
+date: Sun, 28 May 2023 16:56:20 GMT
+content-type: text/html; charset=utf-8
+content-length: 12725
+permissions-policy: interest-cohort=()
+last-modified: Sat, 06 May 2023 08:09:59 GMT
+access-control-allow-origin: *
+strict-transport-security: max-age=31556952
+etag: "64560b57-31b5"
+expires: Sun, 28 May 2023 17:04:05 GMT
+cache-control: max-age=600
+x-proxy-cache: MISS
+x-github-request-id: FD8A:64E8:E68E8:F6229:6473872C
+accept-ranges: bytes
+via: 1.1 varnish
+age: 136
+x-served-by: cache-hkg17920-HKG
+x-cache: HIT
+x-cache-hits: 1
+x-timer: S1685292981.889966,VS0,VE30
+vary: Accept-Encoding
+x-fastly-request-id: 5c0885547b1360efd368516eb77213e7a43586ba
+server: envoy
+x-envoy-upstream-service-time: 1974
+```
+
+
+## Bind Mode and Search Mode
+
+If no filter is specified in its configuration, the middleware runs in the default bind mode, meaning it tries to make a simple bind request to the LDAP server with the credentials provided in the request headers. If the bind succeeds, the middleware forwards the request, otherwise it returns a `401 Unauthorized` status code.
+
+If a filter query is specified in the middleware configuration, and the Authentication Source referenced has a `bindDN` and a `bindPassword`, then the middleware runs in search mode. In this mode, a search query with the given filter is issued to the LDAP server before trying to bind. If result of this search returns only 1 record, it tries to issue a bind request with this record, otherwise it aborts a `401 Unauthorized` status code.
+
+## Config
+
+### Required
+
+- host, string, default "localhost", required
+
+Host on which the LDAP server is running.
+
+- port, number, default 389, required
+
+TCP port where the LDAP server is listening. 389 is the default port for LDAP.
+
+- base_dn, string, "dc=example,dc=com", required
+
+The `baseDN` option should be set to the base domain name that should be used for bind and search queries.
+
+- attribute, string, default "cn", required
+
+Attribute to be used to search the user; e.g., “cn”.
+
+### Optional
+
+- filter, string, default ""
+
+If not empty, the middleware will run in search mode, filtering search results with the given query.
+
+Filter queries can use the `%s` placeholder that is replaced by the username provided in the `Authorization` header of the request. For example: `(&(objectClass=inetOrgPerson)(gidNumber=500)(uid=%s))`, `(cn=%s)`.
+
+- bind_dn, string, default ""
+
+The domain name to bind to in order to authenticate to the LDAP server when running on search mode. Leaving this empty with search mode means binds are anonymous, which is rarely expected behavior. It is not used when running in bind_mode.
+
+- bind_password, string, default ""
+
+The password corresponding to the `bindDN` specified when running in search mode, used in order to authenticate to the LDAP server.
+
+- cache_ttl, number, default 0
```

Comment:
default 0, may not be a good deafult value, that's risk for users, might be default 60s?

Reply:
I believe it would be prudent to have this configuration disabled by default until people explicitly choose to enable it. If the default value is changed to 60, would it be necessary to introduce a switch like start_cache bool?

Reply:
okay, disable cache by default is good.